app/rest/rest/src/main/java/io/syndesis/rest/metrics/PrometheusMetricsProviderImpl.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.metrics;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.internal.readiness.Readiness;
+import io.syndesis.model.metrics.IntegrationMetricsSummary;
+import io.syndesis.rest.metrics.prometheus.HttpClient;
+import io.syndesis.rest.metrics.prometheus.HttpQuery;
+import io.syndesis.rest.metrics.prometheus.QueryResult;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "metrics.kind", havingValue = "prometheus")
+@EnableConfigurationProperties({PrometheusConfigurationProperties.class})
+public class PrometheusMetricsProviderImpl implements MetricsProvider {
+
+    private final String serviceName;
+    private final KubernetesClient kubernetes;
+
+    private final HttpClient httpClient = new HttpClient();
+
+    protected PrometheusMetricsProviderImpl(PrometheusConfigurationProperties config, KubernetesClient kubernetes) {
+        this.serviceName = config.getService();
+        this.kubernetes = kubernetes;
+    }
+
+    @Override
+    public IntegrationMetricsSummary getIntegrationMetricsSummary(String integrationId) {
+
+        // map integration Id to pod name
+        List<Pod> integrationPodList = kubernetes.pods().withLabel("integration").list().getItems();
+        final List<Pod> pods = integrationPodList.stream()
+            .filter(p -> Readiness.isReady(p)
+                && integrationId.equals(p.getMetadata().getAnnotations().get("syndesis.io/integration-id")))
+            .collect(Collectors.toList());
+
+        if (pods.isEmpty()) {
+            throw new IllegalArgumentException("Missing pod with integration id " + integrationId);
+        }
+
+        final String podName = pods.get(0).getMetadata().getName();
```

Comment:
What is if the same integration spans multiple pods ? (e.g if a pod is restarted for some reason). Or if the pod is already dead ?

First I would think that is a valid result that no such pod exists, so throwing an IllegalArgumentException seems to be a bit heavy.

Then, couldn't we avoid to query Kubernetes alltogether ? I mean, the integration pod could expose the integration id in the pod label/annottion, this could be picked up by Prometheus during a scrape run, and the we can query the integration id directly from Prometheus (as it has been stored along with the metrics). 

Not sure though whether Prometheus scraping picks up labels/annotations from pods, but I think this would be very useful (and I would be surprised if this is not already possible).

Reply:
@rhuss you're right, we can avoid using k8s to lookup pods. It looks like k8s can't list terminated pods anyway. 
If the UI can provide `deploymentId` and `integrationId`, we can use that in Prometheus lookup since we are actually using [relabling config](https://github.com/syndesisio/syndesis/blob/master/app/deploy/generator/06-syndesis-prometheus.yml.mustache#L49-L50) to expose _all_ pod labels in metrics.